pkg/alertmanager/state_replication.go

```diff
@@ -134,6 +192,26 @@ func (s *state) Ready() bool {
 	return s.Service.State() == services.Running
 }
 
+// mergeFullStates attempts to merge all full states received from peers during settling.
+func (s *state) mergeFullStates(fs []*clusterpb.FullState) {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	for _, f := range fs {
+		for _, p := range f.Parts {
+			level.Debug(s.logger).Log("msg", "merging full state", "user", s.userID, "key", p.Key, "bytes", len(p.Data))
+
+			if st, ok := s.states[p.Key]; !ok {
+				level.Error(s.logger).Log("msg", "key not found while merging full state", "user", s.userID, "key", p.Key)
+			} else {
```

Comment:
[nit] Can simplify with a guard:

```suggestion
				level.Error(s.logger).Log("msg", "key not found while merging full state", "user", s.userID, "key", p.Key)
				continue
```